subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap.c

```diff
@@ -607,10 +607,12 @@ static void get_shadow_callback(int16_t result_code,
 			memcpy(data->buf, payload, cpy_len);
 		}
 		data->buf[cpy_len] = '\0';
+		data->buf_len = cpy_len;
 	}
 }
 
-int nrf_cloud_coap_shadow_get(char *buf, size_t buf_len, bool delta)
+int nrf_cloud_coap_shadow_get(char *buf, size_t *buf_len, bool delta,
```

Comment:
it looks like the coap endpoint supports: `COAP_CONTENT_FORMAT_APP_OCTET_STREAM`, `COAP_CONTENT_FORMAT_APP_JSON`, and `COAP_CONTENT_FORMAT_APP_CBOR`.

can you add a check for valid `format` values?
adding that info to the function header comments would be good too.

Reply:
Done.